src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -192,34 +213,13 @@ end = struct
     3 + nearest_log_2_incr + nearest_log_2_txn
     + Int.pow 2 (transaction_capacity_log_2 + work_delay_factor)
 
-  let hash t =
-    let state_hash =
-      Parallel_scan.State.hash t.tree
-        (Binable.to_string (module Ledger_proof_with_sok_message))
-        (Binable.to_string (module Transaction_with_witness))
-    in
-    Staged_ledger_aux_hash.of_bytes
-      ((state_hash :> string) ^ Int.to_string t.job_count)
-
   let is_valid t =
     let k = max Constants.work_delay_factor 2 in
     Parallel_scan.parallelism ~state:t.tree
     = Int.pow 2 (Constants.transaction_capacity_log_2 + k)
     && t.job_count < work_capacity ()
     && Parallel_scan.is_valid t.tree
 
-  include Binable.Of_binable
-            (T)
-            (struct
-              type nonrec t = t
-
-              let to_binable = Fn.id
-
-              let of_binable t =
-                assert (is_valid t) ;
```

Comment:
did this assertion make it to your new code?

Reply:
Now it has.